specifyweb/frontend/js_src/lib/components/Merging/definitions.ts

```diff
@@ -18,4 +18,13 @@ export const recordMergingTableSpec: Partial<{
     },
     dialogSpecificText: mergingText.agentContainsGroupDescription(),
   },
+  Locality: {
+    dialogSpecificText: mergingText.localityMerging(),
+  },
+  PaleoContext: {
+    dialogSpecificText: mergingText.paleoContextMerging(),
+  },
+  CollectingEvent: {
+    dialogSpecificText: mergingText.collectingEventMerging(),
```

Comment:
@maxpatiiuk , 

I wanted to generate the strings dynamically using the table config labels but the data model is not instantiated yet when this file in run. Any other idea? 
Tried: 
`dialogSpecificText: `${
      strictGetTable('Locality').label
    } ${mergingText.merging()}`,
  },`
  Issue: this file calls getTables before the data model is being fetch
  
  Also tried: 
```
  dialogSpecificText: `${
     Object.keys(tables).length === 0 ? 'someStaticMessage' : strictGetTable('Locality').label
    } ${mergingText.merging()}`
```

Another question is, this dialogSpecificText was created for when the user tries to merge agent group, at the moment the text should not be displayed for locality - paleo and collecting event correct? So is there really a purpose of having a string here? should we just put an empty string?

Reply:
regardless of what checks or conditions you do, the fact of the matter is that this object is being created in the global scope, so it would be created as soon as the file is loaded (which would be during page load), before datamodel
and object is created just once - values are set only once, not on every access - it's not a function, so `Object.keys(tables).length === 0` doesn't make sense as this expression would be evaluated only once - on page load

if you want the expression to only be evaluated when it's first accessed, then simply turn it into a function:

```ts
  dialogSpecificText: () => mergingText.merging(tables.Locality.label),
```

(and change TypeScript type and usage accordingly to let you do this)

this is exactly what we do in user preferences definitions if you noticed - we are using "Taxon" as a subcategory heading, but datamodel might not be loaded yet

Reply:
you are correct. perharps we should make this a bit more clear

how about we change this type:

```ts
    readonly filterIgnore?: (
      resource: SerializedResource<Tables[TABLE_NAME]>
    ) => SerializedResource<Tables[TABLE_NAME]> | undefined;
    readonly dialogSpecificText: string;
```

to this:

```ts
    readonly unmergable?: {
      matches: (
        resource: SerializedResource<Tables[TABLE_NAME]>
      ) => boolean;
      message: LocalizedString;
    };
```

improvements:
1. specifying the message is not needed unless you also want to filter some records
2. names are more clear (filterIgnore -> unmergable.matches, dialogSpecificText -> message)
3. using LocalizedString over string
4. the function now returns a boolean rather than the resource itself - returning resource is "too powerful" as it's not clear just from function signature whether it can/would modify the resource or not - when it returns a boolean only, it's much more clear that it's just a small, pure, filter functions
5. in the future, if more than one different criteria for "unmergability" would exist, the unmergable object could just be converted into an array of objects to support multiple distinct messages

Reply:
That's exactly what I was going to write! You beat me to it

Reply:
:+1: thank you